integration-cli/docker_cli_inspect_test.go

```diff
@@ -142,33 +121,26 @@ func (s *DockerSuite) TestInspectTypeFlagWithInvalidValue(c *check.C) {
 	dockerCmd(c, "run", "--name=busybox", "-d", "busybox", "true")
 
 	out, exitCode, err := dockerCmdWithError("inspect", "--type=foobar", "busybox")
-	if exitCode != 0 || err != nil {
-		if !strings.Contains(out, "not a valid value for --type") {
-			c.Fatalf("failed to inspect image: %s, %v", out, err)
-		}
-	}
+	c.Assert(err, checker.NotNil, check.Commentf("%s", exitCode))
+	c.Assert(exitCode, checker.Equals, 1, check.Commentf("%s", err))
```

Comment:
I figured more specific was better than less specific, but I cannot say 
why it's return code != 0 and yet not an error.



From:   Vincent Demeester <notifications@github.com>
To:     docker/docker <docker@noreply.github.com>
Cc:     Morgan Bauer/Silicon Valley/IBM@IBMUS
Date:   10/19/2015 22:39
Subject:        Re: [docker] use gocheck asserts instead of fatal (#16784)



In integration-cli/docker_cli_inspect_test.go:
> @@ -142,33 +121,26 @@ func (s *DockerSuite) 
TestInspectTypeFlagWithInvalidValue(c *check.C) {
>                dockerCmd(c, "run", "--name=busybox", "-d", "busybox", 
"true")
> 
>                out, exitCode, err := dockerCmdWithError("inspect", 
"--type=foobar", "busybox")
> -              if exitCode != 0 || err != nil {
> -                              if !strings.Contains(out, "not a valid 
value for --type") {
> -                                              c.Fatalf("failed to 
inspect image: %s, %v", out, err)
> -                              }
> -              }
> +              c.Assert(err, checker.NotNil, check.Commentf("%s", 
exitCode))
> +              c.Assert(exitCode, checker.Equals, 1, 
check.Commentf("%s", err))

I think it safe to say the exitCode we want is always 1, but the previous 
test was just checking it's not 0 so, wondering but nit .
—
Reply to this email directly or view it on GitHub.